dataland-frontend/src/components/resources/frameworkDataSearch/euTaxonomy/EuTaxonomyForNonFinancialsPanel.vue

```diff
@@ -176,6 +202,30 @@ export default defineComponent({
       }
       return kpiValueToFormat;
     },
+
+    /**
+     * Extracts the reports and reporting periods for all data sets.
+     * @param dataAndMetaInfoSets array of data sets includin meta information
+     * @returns array containing an array of company reports and an array of the corresponding reporting periods
+     * as strings
+     */
+    extractReportsAndReportingPeriodsFromDataAndMetaInfoSets(
+      dataAndMetaInfoSets: Array<DataAndMetaInformationEuTaxonomyDataForNonFinancials>,
+    ): [({ [p: string]: CompanyReport } | undefined)[], Array<string>] {
+      const reportingPeriods = [];
+      let tempReportingPeriod: string | undefined;
+      for (const dataAndMetaInfoSet of dataAndMetaInfoSets) {
+        tempReportingPeriod = dataAndMetaInfoSet.metaInfo.reportingPeriod;
+        if (tempReportingPeriod) {
+          reportingPeriods.push(tempReportingPeriod);
+        } else console.log("no reporting period given");
```

Comment:
This `else` should be removed. We don't need console.logs.

Reply:
done